Updated to the new websocket message format

robinsdm opened the pull request.

Commit:
Updated to the new websocket message format

aj-michael commented:
I get this error: [ERROR] /home/a/workspace/simulator/src/main/java/xyz/thepathfinder/simulator/SimulatorMap.java:[35,19] cannot access javax.websocket.DeploymentException
[ERROR] class file for javax.websocket.DeploymentException not found

aj-michael commented:
I get that error on Pathfinder#connect(). I did not get it in 0.0.9. In 0.0.9, #connect did not throw DeploymentExceptions, now it does.

aj-michael commented:
I had to add

```
<dependency>
    <groupId>javax.websocket</groupId>
    <artifactId>javax.websocket-api</artifactId>
    <version>1.1</version>
</dependency>
```

to my pom.xml in order to get the library to work.

I added a commit to this PR that adds that to your build.gradle.

aj-michael commented:
Ok, now the Pathfinder constructor is throwing a NPE...

```
java.lang.reflect.InvocationTargetException
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:297)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: Exception in Application start method
    at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:917)
    at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$156(LauncherImpl.java:182)
    ... 1 more
Caused by: java.lang.NullPointerException
    at xyz.thepathfinder.android.Connection.setServices(Connection.java:70)
    at xyz.thepathfinder.android.Pathfinder.constructPathfinder(Pathfinder.java:110)
    at xyz.thepathfinder.android.Pathfinder.<init>(Pathfinder.java:87)
    at xyz.thepathfinder.simulator.SimulatorMap.start(SimulatorMap.java:38)
    at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$163(LauncherImpl.java:863)
    at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$176(PlatformImpl.java:326)
    at com.sun.javafx.application.PlatformImpl.lambda$null$174(PlatformImpl.java:295)
    at java.security.AccessController.doPrivileged(Native Method)
    at com.sun.javafx.application.PlatformImpl.lambda$runLater$175(PlatformImpl.java:294)
    at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
    at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
    at com.sun.glass.ui.gtk.GtkApplication.lambda$null$50(GtkApplication.java:139)
    ... 1 more
```

aj-michael commented:
Where is the Chimney Swap app?

aj-michael commented:
It looks like on Connection.java line 70 you expect messageHandler to have already been set. But that is only set in an onOpen method that does not appear to get called ever? Is this a race condition waiting for the socket to open?

aj-michael commented:
Constructors are a pretty bad place for race conditions.

aj-michael commented:
I cant figure out how to perform a pickup and dropoff action with your SDK

aj-michael commented:
After on an offline discussion, it was determined that pathfinder-android should not expose javax.websocket-api to the developer because it is an implementation detail. But we will fix it later.

Rebasing commit.

aj-michael committed:
Attempting to fix race condition.

aj-michael committed:
Bumped version.

Commit:
Updated update message to include the last number

Commit:
fixed javadoc

Commit:
Updated the action to return commodities and the allow the user to drop off and pick up commodities and transports.

Commit:
removed ability to change the commodity ids in transports

Commit:
Changed the get sub model methods in cluster to get by path not name

Commit:
removed null pointer exception when the commodities transport isn't found

Commit:
Fixed transport routed messages

aj-michael committed:
IDs are ints, not strings.

Commit:
Creating models should work again

aj-michael committed:
Fix commodities in transport json representation.

Transports have an array of commodity objects, not an array of commodity
ids.

aj-michael committed:
Switched to SLF4J for logging.

This way the developer control what log messages are printed in the same
way that they configure their own app logs.

aj-michael committed:
Fixed IOB Exception.

Commit:
Changed the check in cluster to use subscluster instead of subCluster

Commit:
Merge branch 'topic/websocket-message' of https://www.github.com/CSSE497/pathfinder-android into topic/websocket-message

Commit:
Switched transports to fully use commodities and not their ids

aj-michael commented:
Ok. This works fine for me.
